lua/rustaceanvim/commands/diagnostic.lua

```diff
@@ -234,6 +234,103 @@ local function get_rendered_diagnostic(diagnostic)
+  for _, key in ipairs { 'width', 'height', 'max_width', 'max_height' } do
+    c[key] = nil
+  end
+
+  local width, height = vim.lsp.util._make_floating_popup_size(float_preview_lines, c)
+
+  if user_float_win_config.width and user_float_win_config.width > width then
+    width = user_float_win_config.width
+  end
+  if user_float_win_config.height and user_float_win_config.height > height then
+    height = user_float_win_config.height
+  end
+  if user_float_win_config.max_width and user_float_win_config.max_width < width then
+    float_preview_config.max_width = width
+  end
+  if user_float_win_config.max_height and user_float_win_config.max_height < height then
+    float_preview_config.max_height = height
+  end
+
+  float_preview_config.height = height
+  float_preview_config.width = width
+  return float_preview_config
+end
+
+---@param rendered_diagnostic string
+local function render_ansi_code_diagnostic(rendered_diagnostic)
+  local lines =
+    vim.split(rendered_diagnostic:gsub('[\27\155][][()#;?%d]*[A-PRZcf-ntqry=><~]', ''), '\n', { trimempty = true })
+  local float_preview_lines = vim.deepcopy(lines)
+  table.insert(float_preview_lines, 1, '---')
+  table.insert(float_preview_lines, 1, '1. Open in split')
+  vim.schedule(function()
+    close_hover()
+    local bufnr, winnr = vim.lsp.util.open_floating_preview(
+      float_preview_lines,
+      '',
+      validate_window_dimensions(float_preview_lines, config.tools.float_win_config)
+    )
+    local autocmd_id = vim.api.nvim_create_autocmd('WinEnter', {
+      callback = function(args)
+        if args.buf == bufnr then
+          vim.api.nvim_feedkeys(
+            vim.api.nvim_replace_termcodes(
+              [[<c-\><c-n>]] .. '<cmd>lua vim.api.nvim_win_set_cursor(' .. winnr .. ',{1,0})<CR>',
+              true,
+              false,
+              true
+            ),
+            'n',
+            true
+          )
+        end
+      end,
+    })
+    local chanid = vim.api.nvim_open_term(bufnr, {})
+    vim.api.nvim_create_autocmd('WinClosed', {
+      once = true,
+      pattern = tostring(winnr),
+      callback = function()
+        vim.api.nvim_del_autocmd(autocmd_id)
+      end,
+    })
+    vim.api.nvim_chan_send(chanid, vim.trim('1. Open in split\r\n' .. '---\r\n' .. rendered_diagnostic))
+    _window_state.float_winnr = winnr
+    set_close_keymaps(bufnr)
+    set_open_split_keymap(bufnr, winnr, rendered_diagnostic)
+    if config.tools.float_win_config.auto_focus then
+      vim.api.nvim_set_current_win(winnr)
+      vim.api.nvim_feedkeys(
+        vim.api.nvim_replace_termcodes(
+          '<cmd>lua vim.api.nvim_set_current_win('
+            .. winnr
+            .. ')<CR>'
+            .. [[<c-\><c-n>]]
+            .. '<cmd>lua vim.api.nvim_win_set_cursor('
+            .. winnr
+            .. ',{1,0})<CR>',
+          true,
+          false,
+          true
+        ),
+        'n',
+        true
+      )
```

Comment:
This is to ensure cursor syncs properly, nvim_win_set_cursor is not a synchronized call, when first enter the window, Terminal mode is activated, and cursor it at last line.